src/vis/vis.js

```diff
@@ -137,17 +137,21 @@ var VisModel = Backbone.Model.extend({
     var windshaftClient = new WindshaftClient(windshaftSettings);
 
     // Create the public Analysis Factory
-    this.analysis = new AnalysisService({
+    // TODO: use only AnalysisService static methods without instance
+    this.analysisService = new AnalysisService({
       apiKey: this.get('apiKey'),
       authToken: this.get('authToken'),
       analysisCollection: this._analysisCollection,
       vis: this
     });
 
-    var layersFactory = new LayersFactory({
-      visModel: this,
-      windshaftSettings: windshaftSettings
-    });
+    // Public namespace exposing public methods.
+    this.analysis = {
+      createAnalysis: this.analysisService.createAnalysis,
```

Comment:
Are we changing the public API then?

Reply:
Yes. We are going to remove the analysis method. The question is:
* analysis.createAnalysis() vs
* analysis.create()

Reply:
OUCH! you are right, we might create an alias for **backwards compatibility***.

> *since im using this expression very much im gonna write **BC**

Reply:
I think we don't need backwards compat. We can fix the Builder and DI to use the new method

Reply:
@Jesus89  Remember this methods will be exposed through the engine

#### option1 no namespace

```javascript
// Currently we need a json definition to create an analysis;
const analysisDefinition = `{
  type: 'source',
  id: 'a0',
  query: 'SELECT * from carto.lugares',
}`;
// Use engine to create the analysis
const analysis = engine.createAnalysis(analysisDefinition);
// The analysis will be the source of our layer
const layer1 = new carto.Layer('l1', analysis, style);
```


#### option2 scoped namespace

```javascript
// Currently we need a json definition to create an analysis;
const analysisDefinition = `{
  type: 'source',
  id: 'a0',
  query: 'SELECT * from carto.lugares',
}`;
// Use engine to create the analysis
const analysis = engine.analysis.create(analysisDefinition);
// The analysis will be the source of our layer
const layer1 = new carto.Layer('l1', analysis, style);
```


#### option3  (final version)

```javascript
// The analysis can be instantiated from its constructor
const analysis = new carto.analysis.Source('a1', 'SELECT * from carto.lugares');
// The analysis will be the source of our layer
const layer1 = new carto.Layer('l1', analysis, style);
```

Reply:
We could also leave the "public" method as `analyse` and postpone this decision. There'd be no integration work and we can get back to it once we start exposing analysis management...

Reply:
But the `create` method is different. I prefer to do some integration than use the same method that does different thing, because if we expose `analyse` as the new `create` the behavior is not the same because it doesn't check if the analysis node already exists